### ISSUE TITLE
New Adblock list checker shows last checked time Dec. 20, 1969

### ISSUE THREAD

**Issue by Pentaphon** [bromite/bromite#941]
This is minor so it just needs a screenshot to confirm.  

![time](https://user-images.githubusercontent.com/49374058/108627458-8d0f0980-740a-11eb-80bc-0b9b7f3061ee.png)

**Comment by chromer030** [comment]
Same here.
Sometimes it shows true date , sometimes wrong :

![Screenshot_20210221-184111-01](https://user-images.githubusercontent.com/13361552/108629358-bf555100-7474-11eb-9abc-ec9180e7b9d4.jpeg)

**Comment by NineOutOfTen** [comment]
I'm also seeing ads on websites I visit too. I did use the update button but it didn't work. I did post this as a separate issue but it is closed without comment.

**Comment by csagan5** [comment]
@uazo these are the bugs I could individuate:
* on startup it shows the wrong date and time
* it does not actually do the first update after 5 seconds
* after tapping check now, if you exit the page and then come back, it will still show the wrong date and time; returning to the page one more time will then show the right date and time

So it seems that until the download logic triggers the date is always wrong, and there is a missing update on the UI; perhaps a model observer problem?

**Comment by uazo** [comment]
> perhaps a model observer problem?

I don't know, maybe some event that occurs when user press back button that I manage badly. however I think I can fix it easily, damn I didn't notice before.
however the filters are loaded so I don't think the adblock has any problems.

**Comment by uazo** [comment]
>  I did use the update button but it didn't work.

@NineOutOfTen please post a screenshot

**Comment by NineOutOfTen** [comment]
![Screenshot_20210221-161343.png](https://user-images.githubusercontent.com/60764728/108641989-d0479780-745f-11eb-8ef3-e226bedc3e86.png)

**Comment by NineOutOfTen** [comment]
![Screenshot_20210221-161504.png](https://user-images.githubusercontent.com/60764728/108642014-fe2cdc00-745f-11eb-8eee-a5e6f64ccc45.png)

**Comment by chromer030** [comment]
Same here, ad blocker not working.

**Comment by uazo** [comment]
@csagan5 what kind of regression is it, in your opinion?

technically speaking, there are no changes related to how the filter works, the only one is that before the https request was always made at every start, now only every 7 days or at first installation.

more information is needed, @chromer030 @NineOutOfTen a log extraction would be useful.

**Comment by NineOutOfTen** [comment]
Sir/ma'am I have provided the necessary proof in order to point out a problem with the browser. It is not just I but one other have confirmed that the adblocker with Bromite is not working properly. I have promoted Bromite to other friends and if this isn't fixed soon, the browser's reputation is damaged. Anyone, including yourself, could have confirmed this. I have shown you all the problem, please fix it. Thank you.

**Comment by chromer030** [comment]
@uazo , Is logcat useful for debugging ?, or there is other ways for extracting debug logs, please tell me how to collect logs, i will provide any necessary logs.
thanks.

**Comment by uazo** [comment]
> Is logcat useful for debugging

yes thanks, in the logcat we should have all the informations

**Comment by chromer030** [comment]
@uazo @csagan5 , Before anything else, we appreciate all your efforts in maintaining and improving Bromite project, thanks a million.  “we wish you health, and more than wealth, we wish you love”.

I captured this logcat on starting Bromite and playing this YouTube video : [logcat.txt](https://github.com/bromite/bromite/files/6023166/logcat.txt)

Version: `88.0.4324.187`
Arch: `arm64`
Android version: `7.1.2`
Device model: `Lenovo`

![Screenshot_20210222-190937](https://user-images.githubusercontent.com/13361552/108732374-bfbc1d80-7542-11eb-907d-5eed804aa0ef.png)

**Comment by uazo** [comment]
here the information extracted:

```
02-22 19:09:07.907  5154  5212 I chromium: [INFO:verified_ruleset_dealer.cc(41)] OpenAndSetRulesetFile: /data/user/0/org.bromite.bromite/app_chrome/Subresource Filter/Indexed Rules/27/2.1.16.72969/Ruleset Data is valid: 1
02-22 19:09:08.730  5154  5154 I chromium: [INFO:adblock_updater_service.cc(124)] AdBlockUpdaterService: starting up. First update attempt will take place in 5 seconds. Next update attempt will take place in 604800 seconds. 
02-22 19:09:13.730  5154  5154 I chromium: [INFO:adblock_updater_service.cc(170)] AdBlockUpdaterService: update delayed. Wait 529707 s
```

everything works as expected.
if the problem you indicate is related to advertising on youtube, please take a look https://github.com/bromite/bromite/issues/704 https://github.com/bromite/bromite/issues/648

**Comment by chromer030** [comment]
It's not just about youtube ads, almost on all web sites there is lots of ads , bromite cannot block.
the bug is just on 88.0.4324.187 , previous versions working well and no any ads.

**Comment by csagan5** [comment]
> what kind of regression is it, in your opinion?
> 
> technically speaking, there are no changes related to how the filter works, the only one is that before the https request was always made at every start, now only every 7 days or at first installation.

@uazo is the regression confirmed yet? Could the patch for the [DNS enterprise policy](https://github.com/bromite/bromite/issues/832) be related?

@chromer030 can you mention another example website other than YouTube? What are your secure DNS settings?

**Comment by uazo** [comment]
honestly I had thought about a problem on filters.dat, updated 5 days ago, but I wanted to try it before writing it. in your opinion, can it be?

**Comment by chromer030** [comment]
> example website other than YouTube:

![Screenshot_20210223-095541](https://user-images.githubusercontent.com/13361552/108822467-5d136200-75d4-11eb-8a87-aaaae26a851b.png)


> secure DNS settings 


No matter it's on or off, ads are not blocked, i tested with Secure DNS on Cloudflare and off.

**Comment by drk002** [comment]
I
![Screenshot_2021-02-23-11-39-40-681_org bromite bromite](https://user-images.githubusercontent.com/65331799/108834058-f7bc7280-75cd-11eb-84d5-a6a5a10b1d3f.jpg)
n my Bromite it's ok. It doesn't matter if I enable Secure DNS or not. the view is the same.

**Comment by uazo** [comment]
@csagan5 I checked that page, the adblock works perfectly because it has nothing to do ... looking with the developer tools, nothing is blocked by the contentfilter but yet I don't see any ads.
it's a good case that needs to be investigated, I'm curious.

@chromer030 do you have any active flags (into chrome://flags/) or have you changed the user agent?

**Comment by uazo** [comment]
@chromer030 sorry another thing, are you logged in facebook? can you try with incognito browsing? can you clear all cookies and cache and try again?

**Comment by chromer030** [comment]
@uazo
I'm on default user agent, without active flags, not logged in Facebook, In incognito mode results is same, i cleared cache and data and tried again , same result.
my screen size is : 5.46 inches (403 dpi)

![Screenshot_20210223-172136](https://user-images.githubusercontent.com/13361552/108853600-2b15f600-75fc-11eb-86dd-cdba2362c033.png)

**Comment by csagan5** [comment]
> Interestingly, the size of the screen has something to do with it. if you open it in desktop mode you will see all some ads, but the user agent don't seem to be involved.

This seems a different issue than what @chromer030 is reporting.

> honestly I had thought about a problem on filters.dat, updated 5 days ago, but I wanted to try it before writing it. in your opinion, can it be?

There would be more people reporting the issue if it was that?

@chromer030 please change the URL for the filters to https://www.bromite.org/filters/old_filters.dat and try again

@uazo it is possible that new filters do not cover well ads in his geographical area

**Comment by chromer030** [comment]
@csagan5 , After updating to latest filters throughout settings the issue resolved, no any ads in YouTube or any websites. Thank you.
> new filters do not cover well ads in his geographical area.

Bromite in my device is connected to proxy with diffrent locations , all tests and screenshots are surfed via US ip.

**Comment by csagan5** [comment]
> After updating to latest filters throughout settings the issue resolved, no any ads in YouTube or any websites. Thank you.

@chromer030 which URL are you using? https://www.bromite.org/filters/old_filters.dat or the default https://www.bromite.org/filters/filters.dat ?

**Comment by chromer030** [comment]
@csagan5 , https://www.bromite.org/filters/filters.dat

**Comment by uazo** [comment]
@chromer030 to better understand this strange behavior, can you tell us the exact procedure you followed to solve it?
did you just press the update button? have you changed the url of the filter?

**Comment by chromer030** [comment]
@uazo @csagan5 , Sorry to reporting late, After updating filters to https://github.com/bromite/filters/commit/6bccdafc4dad8e45ea195fb6eb8fa409a5ab8799 in Settings, I thought the issue is resolved because i didn't saw any ads on YouTube or Apkmirror anymore, but now i tested again with different IP locations and DNS settings, the bug persists and the issue is not resolved, sometimes no ads sometimes full of ads, best site to test is apkmirror.com , it's full of ads.

**Comment by csagan5** [comment]
For the ads problem the issue is #960; let's keep the discussion for this bug about #941

**Comment by csagan5** [comment]
Fixed in `88.0.4324.207`.

**Comment by bat999** [comment]
Hi
I still have a problem with "New Adblock list checker shows last checked time..."
I have put some comments here --> https://github.com/bromite/bromite/discussions/966

I'm using Bromite version 88.0.4324.207.
My phone is Xiaomi Redmi Go.
The OS is ANDROID Oreo Go 8.1.0.

![Adblock_Screenshot_2021-03-02_11-48-08](https://user-images.githubusercontent.com/640761/109700919-48ae0700-7b8a-11eb-901b-3c3babb2999e.jpg)

**Comment by SumatraPeter** [comment]
@bat999: Not fixed here either. This issue needs to be reopened.

**Comment by uazo** [comment]
so it is a lottery
@csagan5 I propose a rollback, develop https://github.com/bromite/bromite/issues/944 and then try again

**Comment by csagan5** [comment]
The issue left is that the date is displayed incorrectly? @bat999 @SumatraPeter is this a clean install?

**Comment by SumatraPeter** [comment]
Yep, wrong date/time. My copy's kept updated via your F-Droid repository.

**Comment by bat999** [comment]
> The issue left is that the date is displayed incorrectly? @bat999 @SumatraPeter is this a clean install?

It was installed through F-droid with the extra Bromite repo.
Recently updated from 88.0.4324.187 -> 88.0.4324.207

Now it shows "Last checked 30 Dec 1969 22:20:37"
That's changed from "Last checked 29 Dec 1969 11:19:36"

**Comment by SumatraPeter** [comment]
For me the date now is 01-Jan-1970 while the time keeps changing every time I click the Check Now button (doesn't match my current system TZ time though).

**Comment by uazo** [comment]
> Now it shows "Last checked 30 Dec 1969 22:20:37"
That's changed from "Last checked 29 Dec 1969 11:19:36"

technically, that value is related to how an empty `base::Time()` is internally recorded. unfortunately I don't think it's anything that can help us to understand.

the only thing I can say for now is, if you are able, check how the situation is (see https://github.com/bromite/bromite/issues/960#issuecomment-790687752) and report here. 
the values in the `Local State` file would also be useful.

**Comment by bat999** [comment]
> the only thing I can say for now is, if you are able, check how the situation is (see [#960 (comment)](https://github.com/bromite/bromite/issues/960#issuecomment-790687752)) and report here.
> the values in the `Local State` file would also be useful.

Hi
I don't know how to do this.
Note skilled enough. :-(
Maybe other people can help.

**Comment by csagan5** [comment]
@uazo I can reproduce this with a fresh v89 install. Some logs:

```
03-08 08:55:50.576  2917  2917 I chromium: [INFO:adblock_updater_service.cc(124)] AdBlockUpdaterService: starting up. First update attempt will take place in 5 seconds. Next update attempt will take place in 604800 seconds. 
03-08 08:55:55.978  2917  2917 I chromium: [INFO:download_filters_task.cc(165)] DownloadFiltersTask: fetching URL 'https://www.bromite.org/filters/filters.dat' with method HEAD response_code 200 last_modified 2021-03-01 21:18:40.000 UTC
03-08 08:55:55.978  2917  2917 I chromium: [INFO:download_filters_task.cc(146)] DownloadFiltersTask: Update not needed
03-08 08:56:23.753  2917  2917 I chromium: [INFO:download_filters_task.cc(165)] DownloadFiltersTask: fetching URL 'https://www.bromite.org/filters/filters.dat' with method HEAD response_code 200 last_modified 2021-03-01 21:18:40.000 UTC
03-08 08:56:23.753  2917  2917 I chromium: [INFO:download_filters_task.cc(146)] DownloadFiltersTask: Update not needed
```

No update took place in 5 seconds; I tapped the button instead.

**Comment by uazo** [comment]
have you tried with https://github.com/uazo/bromite/blob/v89/build/patches/Bromite-AdBlockUpdaterService.patch ?

**Comment by csagan5** [comment]
Not yet.

**Comment by uazo** [comment]
i think i solved with that

**Comment by csagan5** [comment]
> i think i solved with that

the patch is based on an old version, not latest v88 release, so I cannot use it. In these cases the git commit should tell what you changed; it looks like there are multiple changes.

I will revert back to an old version of the patch, before the UI was added.

**Comment by csagan5** [comment]
Fixed in `89.0.4389.78`.
